bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/handler/OpenSprinklerDeviceHandler.java

```diff
@@ -36,20 +46,51 @@ public OpenSprinklerDeviceHandler(Thing thing) {
 
     @Override
     protected void updateChannel(ChannelUID channel) {
-        switch (channel.getIdWithoutGroup()) {
-            case SENSOR_RAIN:
-                try {
+        try {
+            switch (channel.getIdWithoutGroup()) {
+                case SENSOR_RAIN:
                     if (getApi().isRainDetected()) {
                         updateState(channel, OnOffType.ON);
                     } else {
                         updateState(channel, OnOffType.OFF);
                     }
-                } catch (CommunicationApiException e) {
-                    logger.debug("Could not update rainsensor", e);
+                    break;
+                case SENSOR_CURRENT_DRAW:
+                    updateState(channel,
+                            new QuantityType<ElectricCurrent>(getApi().currentDraw(), MILLI(SmartHomeUnits.AMPERE)));
+                    break;
+                default:
+                    logger.debug("Not updating unknown channel {}", channel);
+            }
+        } catch (CommunicationApiException | NoCurrentDrawSensorException e) {
+            logger.debug("Could not update {}", channel, e);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        ChannelUID currentDraw = new ChannelUID(thing.getUID(), "currentDraw");
+        ThingBuilder thingBuilder = editThing();
+        if (thing.getChannel(currentDraw) == null) {
+            try {
```

Comment:
No need to edit until you know you might need to.
```suggestion
        if (thing.getChannel(currentDraw) == null) {
            ThingBuilder thingBuilder = editThing();
            try {
```